framework/src/geomsearch/GeometricSearchData.C

```diff
@@ -250,21 +249,18 @@ GeometricSearchData::getQuadratureNearestNodeLocator(const BoundaryName & primar
 }
 
 NearestNodeLocator &
-GeometricSearchData::getQuadratureNearestNodeLocator(const unsigned int primary_id,
-                                                     const unsigned int secondary_id)
+GeometricSearchData::getQuadratureNearestNodeLocator(const BoundaryID primary_id,
+                                                     const BoundaryID secondary_id)
 {
-  // TODO: Make this better!
-  unsigned int base_id = 1e6;
-  unsigned int qsecondary_id = secondary_id + base_id;
+  const auto qsecondary_id = -secondary_id;
```

Comment:
I'm having some second thoughts about the qsecondary ID generation here. I think we should at least have a check to make sure the ID is not used by any boundary. Better yet, update the datastructures to separate the two sets.